content/app-dev/create-provider/new-interface.md

```diff
@@ -0,0 +1,242 @@
+
+Now let's take a look at the data payloads that might be used with these methods. Again, keep in mind that this is an example use case: a real implementation is likely to have far more detail.
+
+#### AuthorizePaymentRequest
+
+The list of fields on the `AuthorizePaymentRequest` message body:
+
+| Field | Description |
+| :--- | :--- |
+| `amount` | Amount of the transaction, in cents |
+| `tax` | Amount of tax applied to this transaction, in cents |
+| `paymentMethod` | Token (string) of the payment method to be used |
+| `paymentEntity` | The entity (customer) requesting this payment |
+| `referenceId` | Opaque Reference ID (e.g. order number) |
+
+More complex implementations of a payment provider might require additional fields for currency and the exchange rate _at the time of transaction_.
+
+#### AuthorizePaymentResponse
+
+The list of fields returned by the payments provider in response to an authorization request:
+
+| Field | Description |
+| :--- | :--- |
+| `success` | Indicates a successful authorization |
+| `authCode` | Optional string containing the transaction ID of the authorization |
+| `failReason` | Optional string with reason for rejection |
+
+#### CompletePaymentRequest
+
+In response to a successful authorization, a payment can be confirmed (real money withdrawn) using the following message body:
+
+| Field | Description |
+| :--- | :--- |
+| `authCode` | Authorization code from previous response |
+| `description` | A description field to be added to the payment summary (e.g., memo field of a credit card statement) |
+
+#### CompletePaymentResponse
+
+The payment capability will return the following message body after processing a payment request:
+
+| Field | Description |
+| :--- | :--- |
+| `success` | Whether the transaction was successful |
+| `txid` | Receipt code / transaction ID |
+| `timestamp` | Timestamp (ms since epoch GMT) of the transaction |
+
+#### PaymentMethodsList
+
+In response to the empty query, a payments capability provider will return the following message body:
+
+| Field | Description |
+| :--- | :--- |
+| `methods` | A map of `string->string` with tokens and descriptions |
+
+
+### Generating the interfaces
+
+Now that we have logically defined the contract to be shared by the payments-consuming actors and the payments capability provider, let's use some of the available wasmCloud tools to generate the actor interface code.
+
+
+```shell
+wash new interface payments
+```
+
```

Comment:
Here, we need to tell the developer to select the `factorial` interface as a template, name it `payments`, and then keep the namespace prefix and contract name default. These are what `wash` will prompt you for